kbs/src/plugins/implementations/resource/backend.rs

```diff
@@ -17,7 +17,11 @@ type RepositoryInstance = Arc<dyn StorageBackend>;
 #[async_trait::async_trait]
 pub trait StorageBackend: Send + Sync {
     /// Read secret resource from repository.
-    async fn read_secret_resource(&self, resource_desc: ResourceDesc) -> Result<Vec<u8>>;
+    async fn read_secret_resource(
+        &self,
+        resource_desc: ResourceDesc,
+        data: &[u8],
```

Comment:
This variable could be more specific. It's not just any data. It's always the request body, right? 

Also, I wonder if you wouldn't be better off just making your own plugin. Plugins already get the entire request body. Might be a better fit. You also probably wouldn't need the first commit.

Reply:
> This variable could be more specific. It's not just any data. It's always the request body, right?
> 

Yes, I can make sure to specify that.

> Also, I wonder if you wouldn't be better off just making your own plugin. Plugins already get the entire request body. Might be a better fit. You also probably wouldn't need the first commit.

Which would you prefer?

Reply:
Probably best to do a plugin; at the very least because you can enable multiple plugins at once, but you can only have one resource backend (maybe not relevant to your use case, but a nice property). Plugins have generally superseded backends. In fact, the resource thing is actually a plugin itself now that only exposes part of what it has to the backend.

Reply:
Ok, I'll edit to make this its own plugin instead.